### PR TITLE
EVG-5687 add distro configuration for weighting factors

### PR DESCRIPTION
**Pull request by StuartButler**

### CODE REVIEW

File: public/static/js/distros.js [hunk 1]
```diff
@@ -9,8 +9,15 @@ mciModule.controller('DistrosCtrl', function($scope, $window, $location, $anchor
   for (var i = 0; i < $scope.distros.length; i++) {
     $scope.distros[i].pool_size = $scope.distros[i].pool_size || 0;
     $scope.distros[i].planner_settings = $scope.distros[i].planner_settings || {};
-    $scope.distros[i].planner_settings.minimum_hosts = $scope.distros[i].planner_settings.minimum_hosts || 0;
     $scope.distros[i].planner_settings.version = $scope.distros[i].planner_settings.version || "legacy";
+    $scope.distros[i].planner_settings.minimum_hosts = $scope.distros[i].planner_settings.minimum_hosts || 0;
+    $scope.distros[i].planner_settings.maximum_hosts = $scope.distros[i].planner_settings.maximum_hosts || 0;
+    $scope.distros[i].planner_settings.target_time = $scope.distros[i].planner_settings.target_time || 0;
```
Comment (john-m-liu) [inline public/static/js/distros.js]: These defaults should be what the defaults are in the legacy scheduler

File: public/static/js/distros.js [hunk 2]
```diff
@@ -9,8 +9,15 @@ mciModule.controller('DistrosCtrl', function($scope, $window, $location, $anchor
   for (var i = 0; i < $scope.distros.length; i++) {
     $scope.distros[i].pool_size = $scope.distros[i].pool_size || 0;
     $scope.distros[i].planner_settings = $scope.distros[i].planner_settings || {};
-    $scope.distros[i].planner_settings.minimum_hosts = $scope.distros[i].planner_settings.minimum_hosts || 0;
     $scope.distros[i].planner_settings.version = $scope.distros[i].planner_settings.version || "legacy";
+    $scope.distros[i].planner_settings.minimum_hosts = $scope.distros[i].planner_settings.minimum_hosts || 0;
+    $scope.distros[i].planner_settings.maximum_hosts = $scope.distros[i].planner_settings.maximum_hosts || 0;
```
Comment (john-m-liu) [inline public/static/js/distros.js]: There's no control for this - is that intentional?

File: service/templates/distros.html [hunk 3]
```diff
@@ -463,15 +463,64 @@ <h2 style="display:inline-block; padding-right:15px">Configure
...
                     role="presentation"><a role="menuitem" tabindex="-1">[[version.display]]</a></li>
                 </ul>
             </div>
-            <div class="icon fa fa-warning distro-error" ng-show="form.plannerVersion.$dirty && form.plannerVersion.$error.required || form.plannerVersion.$invalid">Planner Version is required</div>
+            <div class="icon fa fa-warning distro-error" ng-show="form.plannerSettingsVersion.$dirty && form.plannerSettingsVersion.$error.required ||
+              form.plannerSettingsVersion.$invalid">Planner Version is required</div>
+            <div ng-show="activeDistro.planner_settings.version == 'tunable'">
+              <div>
+                <label class="distro-label">Minimum Hosts:</label>
+                <input ng-readonly="readOnly" type="number" ng-required="activeDistro.planner_settings.version == 'tunable'"
```
Comment (john-m-liu) [inline service/templates/distros.html]: Isn't this a duplicate of the other min hosts input?

File: rest/route/distro_test.go [hunk 4]
```diff
@@ -925,16 +925,28 @@ func (s *DistroPatchByIDSuite) TestRunValidPlannerSettingsVersion() {
...
+		"ERROR: invalid planner_settings.version 'invalid' for distro 'fedora8'",
+		"ERROR: invalid planner_settings.minimum_hosts value of -1 for distro 'fedora8' - the value must be a non-negative integer",
+		"ERROR: invalid planner_settings.maximum_hosts value of -1 for distro 'fedora8' - the value must be a non-negative integer",
+		"ERROR: invalid planner_settings.patch_zipper_factor value of -1 for distro 'fedora8' - the value must be a non-negative integer",
+	}
+
+	error := (resp.Data()).(gimlet.ErrorResponse)
+	for _, err := range errors {
+		s.Contains(error.Message, err)
+	}
```
Comment (tychoish) [inline rest/route/distro_test.go]: This kind of testing is a bit fragile, but it's probably fine. 

I'm fine with this change when John's ok with it.

**Pull request closed without merging**